tests/test_distributed/test_with_spark/test_spark_local.py

```diff
@@ -1124,3 +1093,80 @@ def test_early_stop_param_validation(self):
+                (Vectors.dense(4.5, 5.0, 6.0), 0, 0.29556),
+                (Vectors.dense(9.0, 4.5, 8.0), 0, 2.36570),
+                (Vectors.sparse(3, {1: 1.0, 2: 6.0}), 1, -1.87988),
+                (Vectors.sparse(3, {1: 6.0, 2: 7.0}), 1, -0.30612),
+                (Vectors.sparse(3, {1: 8.0, 2: 10.5}), 1, 2.44826),
+            ],
+            ["features", "qid", "expected_prediction"],
+        )
+        self.ranker_df_train_1 = self.session.createDataFrame(
+            [
+                (Vectors.sparse(3, {1: 1.0, 2: 5.5}), 0, 9),
+                (Vectors.sparse(3, {1: 6.0, 2: 7.5}), 1, 9),
+                (Vectors.sparse(3, {1: 8.0, 2: 9.5}), 2, 9),
+                (Vectors.dense(1.0, 2.0, 3.0), 0, 8),
+                (Vectors.dense(4.0, 5.0, 6.0), 1, 8),
+                (Vectors.dense(9.0, 4.0, 8.0), 2, 8),
+                (Vectors.sparse(3, {1: 1.0, 2: 5.5}), 0, 7),
+                (Vectors.sparse(3, {1: 6.0, 2: 7.5}), 1, 7),
+                (Vectors.sparse(3, {1: 8.0, 2: 9.5}), 2, 7),
+                (Vectors.dense(1.0, 2.0, 3.0), 0, 6),
+                (Vectors.dense(4.0, 5.0, 6.0), 1, 6),
+                (Vectors.dense(9.0, 4.0, 8.0), 2, 6),
+                (Vectors.sparse(3, {1: 1.0, 2: 5.5}), 0, 5),
+                (Vectors.sparse(3, {1: 6.0, 2: 7.5}), 1, 5),
+                (Vectors.sparse(3, {1: 8.0, 2: 9.5}), 2, 5),
+                (Vectors.dense(1.0, 2.0, 3.0), 0, 4),
+                (Vectors.dense(4.0, 5.0, 6.0), 1, 4),
+                (Vectors.dense(9.0, 4.0, 8.0), 2, 4),
+                (Vectors.sparse(3, {1: 1.0, 2: 5.5}), 0, 3),
+                (Vectors.sparse(3, {1: 6.0, 2: 7.5}), 1, 3),
+                (Vectors.sparse(3, {1: 8.0, 2: 9.5}), 2, 3),
+                (Vectors.dense(1.0, 2.0, 3.0), 0, 2),
+                (Vectors.dense(4.0, 5.0, 6.0), 1, 2),
+                (Vectors.dense(9.0, 4.0, 8.0), 2, 2),
+                (Vectors.sparse(3, {1: 1.0, 2: 5.5}), 0, 1),
+                (Vectors.sparse(3, {1: 6.0, 2: 7.5}), 1, 1),
+                (Vectors.sparse(3, {1: 8.0, 2: 9.5}), 2, 1),
+                (Vectors.dense(1.0, 2.0, 3.0), 0, 0),
+                (Vectors.dense(4.0, 5.0, 6.0), 1, 0),
+                (Vectors.dense(9.0, 4.0, 8.0), 2, 0),
+            ],
+            ["features", "label", "qid"],
+        )
+
+    def test_ranker(self):
+        ranker = SparkXGBRanker(qid_col="qid")
+        assert ranker.getOrDefault(ranker.objective) == "rank:pairwise"
+        model = ranker.fit(self.ranker_df_train)
+        pred_result = model.transform(self.ranker_df_test).collect()
+
+        for row in pred_result:
+            assert np.isclose(row.prediction, row.expected_prediction, rtol=1e-3)
+
+    def test_ranker_qid_sorted(self):
+        ranker = SparkXGBRanker(qid_col="qid", num_workers=2)
+        assert ranker.getOrDefault(ranker.objective) == "rank:pairwise"
+        model = ranker.fit(self.ranker_df_train_1)
+        model.transform(self.ranker_df_test).collect()
```

Comment:
what's the purpose of this test?

Reply:
to test if the SparkRanker will throw exception